pkg/ui/userinterface.go

```diff
@@ -58,6 +60,40 @@ func (ui *consoleUi) Output(output string) error {
 }
 
 func (ui *consoleUi) OutputError(err error) error {
+	// nothing needs to be done if err is nil
+	if err == nil {
+		return nil
+	}
+
+	// for simplistic handling of error catalog errors
+	var snykError snyk_errors.Error
+	if errors.As(err, &snykError) {
+		mainPattern := "%-8s %s"
+		titlePattern := "%s (%s)"
+
+		uiError := utils.ErrorOf(fmt.Fprintln(ui.errorWriter, fmt.Sprintf(mainPattern, strings.ToUpper(snykError.Level), fmt.Sprintf(titlePattern, snykError.Title, snykError.ID))))
+		if uiError != nil {
+			return uiError
+		}
+
+		if len(snykError.Detail) > 0 {
+			uiError = utils.ErrorOf(fmt.Fprintln(ui.errorWriter, fmt.Sprintf(mainPattern, "Info:", snykError.Detail)))
+			if uiError != nil {
+				return uiError
+			}
+		}
+
+		for _, l := range snykError.Links {
+			uiError = utils.ErrorOf(fmt.Fprintln(ui.errorWriter, fmt.Sprintf(mainPattern, "Help:", l)))
+			if uiError != nil {
+				return uiError
+			}
+		}
+
+		return nil
+	}
+
+	// Default handling for all other errors
```

Comment:
**suggestion:** Introduce some tests to cover the various branches we have within the OutputError method.

Reply:
Done!